src/MySqlConnector/Core/SchemaProvider.cs

```diff
@@ -33,6 +34,35 @@ public DataTable GetSchema(string collectionName)
 			return dataTable;
 		}
 
+		public DataTable GetSchema(string collectionName, string[] restrictions)
+		{
+			Action close = null;
+			if (m_connection.State != ConnectionState.Open)
+			{
+				m_connection.Open();
+				close = m_connection.Close;
+			}
+
+			var dataTable = m_connection.GetSchema(collectionName, restrictions);
```

Comment:
I think this will introduce infinite recursion between this method and `MySqlConnection.GetSchema`.

Please add tests that verify the correct information is being returned, for example like https://github.com/mysql-net/MySqlConnector/blob/5758207a81fad9c77494995316fcfc4687f04bb7/tests/SideBySide/QueryTests.cs#L1147-L1148

Reply:
Yeap! I will add all the tests once I get things right. I am just confused on how to get the restrictions now if this will lead to a infinite recursion. Could you shed some light, please?

Reply:
`MySqlConnection.GetSchema` calls this method: https://github.com/mysql-net/MySqlConnector/blob/a6b5d69c6640d5de0b15051f6e734bb4ab75c962/src/MySqlConnector/MySql.Data.MySqlClient/MySqlConnection.cs#L311

(Or it will do, once the overload has changed.)

So this method shouldn't call back to `MySqlConnection.GetSchema`; it should just do the work.